apecs-physics/src/Apecs/Physics/Collision.hs

```diff
@@ -117,26 +120,39 @@ destroyCollisionHandler = error "Destroy CollisionHandler not yet implemented"
 instance Component CollisionHandler where
   type Storage CollisionHandler = Space CollisionHandler
 
-instance Has w IO Physics => Has w IO CollisionHandler where
+instance (MonadIO m, Has w m Physics) => Has w m CollisionHandler where
   getStore = (cast :: Space Physics -> Space CollisionHandler) <$> getStore
 
-instance ExplSet IO (Space CollisionHandler) where
-  explSet sp@(Space _ _ _ hMap spcPtr) ety handler = do
+instance (MonadIO m) => ExplSet m (Space CollisionHandler) where
+  explSet sp@(Space _ _ _ hMap spcPtr) ety handler = liftIO $ do
     explDestroy sp ety
     hPtr <- newCollisionHandler spcPtr handler ety
     modifyIORef' hMap (M.insert ety (Record hPtr handler))
 
-instance ExplDestroy IO (Space CollisionHandler) where
-  explDestroy (Space _ _ _ hMap _) ety = do
+instance (MonadIO m) => ExplDestroy m (Space CollisionHandler) where
+  explDestroy (Space _ _ _ hMap _) ety = liftIO $ do
     rd <- M.lookup ety <$> readIORef hMap
     forM_ rd$ \(Record c _) -> destroyCollisionHandler c >> modifyIORef' hMap (M.delete ety)
 
-instance ExplMembers IO (Space CollisionHandler) where
-  explMembers (Space _ _ _ hMap _) = U.fromList . M.keys <$> readIORef hMap
+instance (MonadIO m) => ExplMembers m (Space CollisionHandler) where
+  explMembers (Space _ _ _ hMap _) = liftIO $ U.fromList . M.keys <$> readIORef hMap
 
-instance ExplGet IO (Space CollisionHandler) where
-  explExists (Space _ _ _ hMap _) ety = M.member ety <$> readIORef hMap
-  explGet (Space _ _ _ hMap _) ety = do
+instance (MonadIO m) => ExplGet m (Space CollisionHandler) where
+  explExists (Space _ _ _ hMap _) ety = liftIO $ M.member ety <$> readIORef hMap
+  explGet (Space _ _ _ hMap _) ety = liftIO $ do
     Just (Record _ handler) <- M.lookup ety <$> readIORef hMap
     return handler
 
+addPostStepCallback :: (Has w m Physics, MonadIO m) => Int -> SystemT w IO () -> SystemT w m ()
```

Comment:
Would you mind adding some haddocks on how to use this? Or maybe just provide a link to https://chipmunk-physics.net/release/ChipmunkLatest-Docs/#CollisionCallbacks-PostStep